scip-java/src/main/resources/scip-java/scip_java.bzl

```diff
@@ -70,15 +70,15 @@ def _scip_java(target, ctx):
     output_dir = []
 
     for source_jar in source_jars:
-        dir = ctx.actions.declare_directory("extracted_srcjar/" + source_jar.short_path)
+        dir = ctx.actions.declare_directory(ctx.label.name + "/extracted_srcjar/" + source_jar.short_path)
         output_dir.append(dir)
     
         ctx.actions.run_shell(
             inputs = javac_action.inputs,
             outputs = [dir],
             mnemonic = "ExtractSourceJars",
             command = """
-                unzip {input_file} -d {output_dir}
+                [ "$(unzip -q -l {input_file} | wc -l)" -eq 0 ] || unzip {input_file} -d {output_dir}
```

Comment:
Inexplicably, unzip returns a 1 exit code if there are any _warnings_ (see Diagnostics on https://linux.die.net/man/1/unzip) with no way to turn it off.

So we just literally count the number of files..